### ISSUE TITLE
Proper support using custom icon packs

### ISSUE THREAD

**Issue by BryceBarbara** [primefaces/primevue#1627]
**I'm submitting a ...**
```
[x] feature request
```
This story is about adding ability to use custom icon providers like [iconify](https://iconify.design/) rather than only having primeicons.

This isn't related to the fact that some components allow you to provide a template _manually every time_ to use custom icons. But some components (things like menus) don't provide that option at all.

```html
<Button type="button" class="p-button-outlined p-button-success">
    <Icon icon="{some-custom-icon-here}" style="width: 1.5rem" />
    <span class="p-ml-2 p-text-bold">PrimeVue</span>
</Button>
```
compared to

```html
<Button type="button" class="p-button-outlined p-button-success" icon="{some-custom-icon-here}" label="PrimeVue" />
```

**Comment by JohnCampionJr** [comment]
Would definitely like this too but I suspect its a rather major change.

**Comment by BryceBarbara** [comment]
@JohnCampionJr Yeah, it's bit of a major change. I created a PR that implements the major changes for the team to use as a good starting point although it could be accepted as-is.
